pkg/client/apifactory.go

```diff
@@ -203,6 +206,10 @@ func (s *APIFactory) Start() {
 	// launch clients
 	if !s.IsTestingMode() {
 		s.clients.Run(s.stopChan)
+		if err := s.WaitForSync(); err != nil {
```

Comment:
We already know we're not in test mode why are we not calling `s.clients.WaitForSync(time.Second, 30*time.Second)` directly?

Reply:
Changed it.